src/qw/cli.py

```diff
@@ -1,30 +1,118 @@
+"""
+The qw (Quality Workflow) tool helps enforce regulatory compliance
+for projects managed on github.
+"""
+
+import json
+import os
+import sys
 from typing import Annotated
 
-import typer as typer
+import git
+import typer
 from loguru import logger
 
-# Main entry point into the application
+import qw.service
+from qw.base import QwError
+
 app = typer.Typer()
 
 
 @app.command()
-def parrot(
-        example: Annotated[str, typer.Argument(help="String to print to user")]
+def init(
+    repo: Annotated[
+        str | None,
+        typer.Option(
+            help="The URL (or remote name) for the repo containing the issues."
+            + " If not supplied the remotes named 'upstream' and 'origin' will"
+            + " be tried.",
+        ),
+    ] = None,
+    service: Annotated[
+        qw.service.Service,
+        typer.Option(
+            help="Which service is hosting the issue tracker. Not required"
+            + " if the repo URL begins 'github' or 'gitlab'.",
+        ),
+    ] = None,
+    force: Annotated[
+        bool,
+        typer.Option(
+            help="Replace any existing configuration.",
+        ),
+    ] = False,
 ) -> None:
-    parrot_name = typer.prompt("What's your parrot's name?", default="Polly")
-    shout = typer.confirm("Do you want your parrot to shout?")
-    output_message = example
-    if shout:
-        output_message = f"{output_message.upper()}!!"
-    logger.info("{parrot_name}🦜: {output_message}", parrot_name=parrot_name, output_message=output_message)
+    """Initializes this tool and the repository (as far as possible)."""
+    git_dir = qw.service.find_aunt_dir(".git")
+    if not git_dir:
+        msg = "We are not in a git project, so we cannot initialize!"
+        raise QwError(
+            msg,
+        )
+    base = os.path.dirname(git_dir)
+    gitrepo = git.Repo(base)
+    repo = qw.service.get_repo_url(gitrepo, repo)
+    qw_dir = os.path.join(base, ".qw")
+    logger.info(
+        ".qw directory is '{dir}'",
+        dir=qw_dir,
+    )
+    if os.path.isfile(qw_dir):
+        msg = ".qw file exists, which is blocking us from making a .qw directory. Please delete it!"
+        raise QwError(
+            msg,
+        )
+    elif not os.path.isdir(qw_dir):
+        os.mkdir(qw_dir)
+    elif not force:
+        msg = ".qw directory already exists! Use use existing configuration or use --force flag to reinitialize!"
+        raise QwError(
+            msg,
+        )
+    hur = qw.service.remote_address_to_host_user_repo(repo)
```

Comment:
Could unpack this directly and have the exception raising in the `remote_address_to_host_user_repo` method?
```suggestion
   host, owner, repo = qw.service.remote_address_to_host_user_repo(repo)
```

Reply:
Yes! Should have cleaned that up when I shifted the exception. Nice spot.